packages/coil-extension/src/content/services/RunContentHandler.ts

```diff
@@ -54,6 +77,47 @@ export class RunContentHandler {
     }
   }
 
+  private async pollForYouTubeChannelId({
+    everyMs,
+    times
+  }: PollForYouTubeChannelIdParams) {
+    try {
+      let polls = 0
+      while (++polls <= times) {
+        const channelElem = this.document.querySelector<HTMLAnchorElement>(
+          '.ytd-channel-name > a.yt-simple-endpoint'
+        )
+        if (channelElem) {
+          const channelUrl = channelElem.href
+          debug('channelUrl', channelUrl)
+          if (channelUrl.match(/^\/channel\/\w+/)) {
+            return channelUrl
+          } else {
+            break
+          }
+        } else {
+          await new Promise(resolve => setTimeout(resolve, everyMs))
+        }
+      }
+    } catch (err) {
+      debug('Failed to grab channel URL from Youtube page. err=', err)
+    }
+  }
+
+  async windowLoaded() {
+    if (this.document.readyState !== 'complete') {
+      await new Promise(resolve => {
+        this.window.addEventListener(
+          'load',
+          () => {
+            resolve()
+          },
+          { once: true }
```

Comment:
@njlie 
Probably doesn't *really* matter but could potentially add a bunch of persistent listeners given youtube uses HTML5 pushState so can use once:true